components/integrations/abstract_command.tsx

```diff
@@ -196,61 +188,64 @@ export default class AbstractCommand extends React.PureComponent {
         this.props.action(command).then(() => this.setState({saving: false}));
     }
 
-    updateDisplayName = (e) => {
+    updateDisplayName = (e: ChangeEvent<HTMLInputElement>) => {
         this.setState({
             displayName: e.target.value,
         });
     }
 
-    updateDescription = (e) => {
+    updateDescription = (e: ChangeEvent<HTMLInputElement>) => {
         this.setState({
             description: e.target.value,
         });
     }
 
-    updateTrigger = (e) => {
+    updateTrigger = (e: ChangeEvent<HTMLInputElement>) => {
         this.setState({
             trigger: e.target.value,
         });
     }
 
-    updateUrl = (e) => {
+    updateUrl = (e: ChangeEvent<HTMLInputElement>) => {
         this.setState({
             url: e.target.value,
         });
     }
 
-    updateMethod = (e) => {
-        this.setState({
-            method: e.target.value,
-        });
+    updateMethod = (e: ChangeEvent<HTMLSelectElement>) => {
+        const methodValue = e.target.value;
+        if (methodValue === 'P' || methodValue === 'G' || methodValue === '') {
+            this.setState({
+                method: methodValue,
+            });
+        }
```

Comment:
lets revert to previous and then you can use `methodValue as State['method']`. Or extract the type of method to another type and use it. eg
```ts
type RequestMethod = 'P' ......
```